engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2405,6 +2540,46 @@ private LogicalOperatorInfo fillLogicalOperatorInfo(OperatorMeta operator)
     return loi;
   }
 
+  private LogicalModuleInfo fillLogicalModuleInfo(ModuleMeta module, boolean flatten, String parentModuleName)
+  {
+    LogicalModuleInfo lmi = new LogicalModuleInfo();
+    lmi.name = module.getName();
+    lmi.className = module.getModule().getClass().getName();
+    if (flatten) {
+      for (OperatorMeta operatorMeta : getLogicalPlan().getAllOperators()) {
```

Comment:
module.getDag().getAllOperator() will return the operatorMeta for the operator which are populated by this module also it will contains operatorMeta populated by the inner modules. The operatorMeta reference is not shared between inner module and outer module. when inner module DAG is merged with the outmodue dag, outer module dag create a new operatorMeta also sets it moduleName from where this operator was added. The Operator object reference are shared.